### PR TITLE
ARROW-8114: [Java][Integration] Enable custom_metadata integration test

### PR DESCRIPTION
**Pull request by rymurr**
This fixes custom metadata on the Java side by correcting the type used in JSON deserialisation

### CODE REVIEW

**Review by siddharthteotia — APPROVED** [review]

File: java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Schema.java [hunk 1]
```diff
@@ -99,21 +101,38 @@ public static Schema convertSchema(org.apache.arrow.flatbuf.Schema schema) {
   private final Map<String, String> metadata;
 
   public Schema(Iterable<Field> fields) {
-    this(fields, null);
+    this(fields, (Map<String, String>) null);
   }
 
   /**
```
Comment (lidavidm) [inline java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Schema.java:107]: nit: seems like this doc comment got moved away from the constructor it's supposed to target.

Comment (rymurr) [reply java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Schema.java]: thanks @lidavidm just fixed.

Comment (lidavidm) [reply java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Schema.java]: Ah, that set off checkstyle, sorry :( (looks like it still wants a docstring for the public constructor)

Comment (rymurr) [reply java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Schema.java]: thats what I get for commiting w/o building ;-) 

fixed again.

Comment (lidavidm) [reply java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Schema.java:107]: Thanks! I'll merge once CI is green.

**Pull request closed without merging**